src/gsi/read_radarref_mosaic_directDA.f90

```diff
@@ -0,0 +1,490 @@
+!          been interpolated into analysis grid and in form of BUFR.
+!
+! PROGRAM HISTORY LOG:
+!    2008-12-20  Hu  make it read in BUFR form reflectivity  data
+!    2010-04-09  Hu  make changes based on current trunk style
+!    2013-03-27  Hu  add code to map obs from WRF mass H grid to analysis grid
+!    2015-02-23  Rancic/Thomas - add l4densvar to time window logical
+!    2015-03-23  Su  -fix array size with maximum message and subset number from
+!                        fixed number to dynamic allocated array
+!    2016-09-xx  CAPS(G. Zhao) 
+!                     - modified Ming Hu's read_radarref_mosaic for 
+!                     - reading mosaic ref dbz for GSIv3.5 variational and
+!                     - enkf analysis, not for gsd-cloud-analysis
+!    2016-11-xx  CAPS(G. Zhao)
+!                     - fix the bugs:
+!                     - dbz bufr with scale=2 to have two more digits accuracy
+!                     - station id
+!                     - obs lat/lon in degree
+!
+!   input argument list:
+!     infile   - unit from which to read mosaic information file
+!     obstype  - observation type to process
+!     lunout   - unit to which to write data for further processing
+!     twind    - input group time window (hours)
+!     sis      - observation variable name
+!
+!   output argument list:
+!     nread    - number of type "obstype" observations read
+!     ndata    - number of type "obstype" observations retained for further processing
+!     nobs     - array of observations on each subdomain for each processor
+!
+! USAGE:
+!   INPUT FILES:  refInGSI
+!
+!   OUTPUT FILES:
+!
+! REMARKS:
+!
+! ATTRIBUTES:
+!   LANGUAGE: FORTRAN 90
+!   MACHINE:  Linux cluster(Wjet)
+!
+!$$$
+!
+!_____________________________________________________________________
+!
+  use kinds, only: r_kind,r_double,i_kind
+  use constants, only: zero,one,rad2deg,deg2rad
+  use convinfo, only: nconvtype,ctwind,cgross,cermax,cermin,cvar_b,cvar_pg, &
+        ncmiter,ncgroup,ncnumgrp,icuse,ictype,icsubtype,ioctype
+  use gsi_4dvar, only: l4dvar,l4densvar,winlen
+  use gridmod, only: nlon,nlat,nlon_regional,nlat_regional,txy2ll,tll2xy
+  use mod_wrfmass_to_a, only: wrfmass_obs_to_a8
+  use mpimod, only: npe
+  use obsmod, only: perturb_obs
+  use directDA_radaruse_mod, only: oe_dbz, refl_lowbnd_dbz
+
+  implicit none
+!
+
+  character(10),    intent(in)    :: infile,obstype
+  integer(i_kind),  intent(in)    :: lunout
+  integer(i_kind),  intent(inout) :: nread,ndata
+  integer(i_kind),  intent(inout) :: nodata
+  integer(i_kind),dimension(npe) ,intent(inout) :: nobs
+  real(r_kind),     intent(in   ) :: twind
+  character(20),    intent(in)    :: sis
+
+! Declare local parameters
+  real(r_kind),parameter:: r90  = 90.0_r_kind
+  real(r_kind),parameter:: r360 = 360.0_r_kind
+
+!
+!  For reflectiivty mosaic
+!
+  integer(i_kind) nreal,nchanl
+
+  integer(i_kind) ifn,i
+ 
+  real(r_kind)  :: maxref
```

Comment:
The variable, maxref, has been identified as unused.  Please remove unused variable.

Reply:
We removed that variable as you suggested.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-1833e300b7dd38188cb415090239f3f184b0ad14deaae73fb732e4fa73d4f221R86